adapter/src/dummy.rs

```diff
@@ -18,31 +18,18 @@ pub struct DummyAdapter {
 // check if a channel is valid
 impl ChannelValidator for DummyAdapter {}
 
-impl Adapter for DummyAdapter {
-    type Output = DummyAdapter;
-
-    fn init(opts: AdapterOptions, config: &Config) -> AdapterResult<DummyAdapter> {
-        let (identity, authorization_tokens, session_tokens) = match opts {
-            AdapterOptions::DummAdapter {
-                dummy_identity,
-                dummy_auth,
-                dummy_auth_tokens,
-            } => (dummy_identity, dummy_auth, dummy_auth_tokens),
-            _ => {
-                return Err(AdapterError::Configuration(
-                    "dummy_identity, dummy_auth, dummy_auth_tokens required".to_string(),
-                ))
-            }
-        };
-
-        Ok(Self {
-            identity,
+impl DummyAdapter {
+    pub fn init(opts: DummyAdapterOptions, config: &Config) -> Self {
```

Comment:
`DummyAdapter::init` has no failing case, if we pass it's options directly, no need to return a `Result`